File: euler_50_consecutive_prime_sum.java
import java.util.*;

public class euler_50_consecutive_prime_sum {
    private int maxPrimeSum; //Declare max prime sum
    private int maxTermCount; //Declare max term count

    //Constructor for declared attributes
    public euler_50_consecutive_prime_sum() {
        this.maxPrimeSum = 0;
        this.maxTermCount = 0;
    }

    //Generate all prime numbers up to the nth term
    public static List<Integer> sieveOfEratosthenes(int n) {
        boolean[] primes = new boolean[n+1]; //Create boolean array
        Arrays.fill(primes, true); //Fill the boolean array with true at all indexes

        //Loop from numbers 2 to the square root of n
        for (int p = 2; p * p <= n; p++) {

            //Check if primes at index p is true (is a prime)
            if (primes[p]) {
                int j = p * p; //Start at p squared

                //While j is less than or equal to n
                while (j <= n) {
                    primes[j] = false; //Set primes at index j to false (not a prime)
                    j += p; //Add p to j (every multiple is not a prime)
                }
            }
        }
        List<Integer> primeNumbers = new ArrayList<>(); //Create a new integer array to hold the prime numbers
        primeNumbers.add(2); //Add 2 as it's the only even prime

        //Loop through every odd before or equal to n
        for (int p = 3; p <= n; p += 2) {

            //If primes at index p is true (is a prime)
            if (primes[p]) {
                primeNumbers.add(p); //Add index p to the int list
            }
        }
        return primeNumbers; //Return primes
    }

    public void findMaxConsecutivePrimeSum(int n) {
        List<Integer> primeNumbers = sieveOfEratosthenes(n); //Generate primes

        //Iterate through every prime in the array
        for (int prime : primeNumbers) {

            //Iterate through every starting prime
            for (int start = 0; primeNumbers.get(start) < prime; start++) {
                int primeSum = 0;
                int termCount = 0;

                //Iterate through every prime that comes after the start prime
                for (int p = start; p < primeNumbers.size(); p++) {
                    primeSum += primeNumbers.get(p);
                    termCount++; //Increment term count

                    //Break and move on to next start pos if the consecutive sum goes over the prime
                    if (primeSum > prime) {
                        break;
                    }

                    //If the consecutive prime sum is equal to the prime
                    if (primeSum == prime) {

                        //Check if the number of primes in the consecutive sum exceeds the current highest
                        if (termCount > this.maxTermCount) {
                            this.maxPrimeSum = primeSum; //Change max prime sum attribute
                            this.maxTermCount = termCount; //Change max term count attribute
                        }
                    }
                }
            }
        }
    }

    public static void main(String[] args) {
        euler_50_consecutive_prime_sum primeSum = new euler_50_consecutive_prime_sum(); //Create a new prime sum object
        primeSum.findMaxConsecutivePrimeSum(Integer.parseInt(args[0])); //Calculate the max consecutive prime sum
        System.out.println(primeSum.maxPrimeSum);
    }
}

File: euler-50-consecutive-prime-sum.go
package main

import(
	"fmt"
	"os"
	"strconv"
)

func main() {
	arg, _ := strconv.Atoi(os.Args[1]) // Get command-line argument
	fmt.Println(find_max_consecutive_prime_sum(arg)) // Print the max consecutive prime sum
}

//Sieve of eratosthenes algorithm finds all prime numbers below or equal to n
//https://www.baeldung.com/cs/prime-number-algorithms
func sieve_of_eratosthenes(n int) []int {
	primes := make([]bool, n+1) // Create a new slice to hold primes
	for i := range primes {
		primes[i] = true // Set every index of primes to True
	}
	// Loop over the numbers from 2 to the square root of n
	for p := 2; p * p <= n; p++ {
		if primes[p] {
			j := p * p // Start from the square of the current prime number
			for j <= n { // While j is less than or equal to n
				primes[j] = false
				j += p // Move to next multiple
			}
		}
	}
	primeNumbers := []int{2} // Creates new list to hold prime numbers, 2 is pre-placed inside as it's the only even prime
		// Loops the length of the boolean list we created with the sieve of eratosthenes, loop every other index
		for p := 3; p <= n; p += 2 {
			if primes[p] { // If the list entry is true, then that index is a prime number
				primeNumbers = append(primeNumbers, p) // Append the index as a prime in the new list
			}
		}
	return primeNumbers
}

func find_max_consecutive_prime_sum(n int) int {
	primeNumbers := sieve_of_eratosthenes(n) // Get list of prime numbers
	max_prime_sum := 0 // Variable to keep the prime with the most consecutive terms that it's the sum of
	max_term_count := 0 // Variable to keep the term count of the highest prime

	for _, prime := range primeNumbers { // For loop that visits every prime in the list
		for start := 0; primeNumbers[start] < prime; start++ { // Loop that changes what prime to start adding from, ex. start from 3 is finished, new start is 5
			prime_sum := 0 // Variable that holds the sum of the primes visited so far
			term_count := 0 // Variable that keeps count of number of primes added up
			for p := start; p < len(primeNumbers); p++ { // For loop goes through every prime after the start prime
				prime_sum += primeNumbers[p] // Add prime to the total sum
				term_count++ // Increment term count
				if prime_sum > prime { // Check if the sum has exceeded the prime we are checking, if it has, we break out of the loop and start adding from a new prime
					break
				}
				if prime_sum == prime { // Check if the sum is equal to prime, if true that means we found a consecutive sum
					if term_count > max_term_count { // Check if the newfound consecutive sum has a higher term count than any we found already
						max_prime_sum = prime_sum // New highest prime
						max_term_count = term_count // New highest term count
					}
				}
			}
		}
    }
	return max_prime_sum
}
